src/ResourceManager/Kubernetes/Commands.Kubernetes/Commands/CreateOrUpdateKubeBase.cs

```diff
@@ -0,0 +1,363 @@
+        public string SshKeyValue { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public Hashtable Tags { get; set; }
+
+        protected ManagedCluster BuildNewCluster()
+        {
+            if (!string.IsNullOrEmpty(ResourceGroupName) && string.IsNullOrEmpty(Location))
+            {
+                var rg = RmClient.ResourceGroups.Get(ResourceGroupName);
+                Location = rg.Location;
+                WriteVerbose(string.Format("Using location {0} from the resource group {1}.", Location,
+                    ResourceGroupName));
+            }
+
+            if (string.IsNullOrEmpty(DnsNamePrefix))
+            {
+                DnsNamePrefix = DefaultDnsPrefix();
+            }
+
+            WriteVerbose(string.Format("Using DNS name prefix {0}.", DnsNamePrefix));
+            SshKeyValue = GetSshKey(SshKeyValue);
+
+            var defaultAgentPoolProfile = new ContainerServiceAgentPoolProfile(
+                "default",
+                NodeVmSize,
+                NodeCount,
+                NodeOsDiskSize,
+                DnsNamePrefix);
+
+            var pubKey =
+                new List<ContainerServiceSshPublicKey> {new ContainerServiceSshPublicKey(SshKeyValue)};
+
+            var linuxProfile =
+                new ContainerServiceLinuxProfile(AdminUserName,
+                    new ContainerServiceSshConfiguration(pubKey));
+
+            var acsServicePrincipal = EnsureServicePrincipal(ClientId, ClientSecret);
+
+            var spProfile = new ContainerServiceServicePrincipalProfile(
+                acsServicePrincipal.SpId,
+                acsServicePrincipal.ClientSecret);
+
+            WriteVerbose(string.Format("Deploying your managed Kubernetes cluster.", AcsSpFilePath));
+            var managedCluster = new ManagedCluster(
+                Location,
+                name: Name,
+                tags: TagsConversionHelper.CreateTagDictionary(Tags, true),
+                dnsPrefix: DnsNamePrefix,
+                kubernetesVersion: KubernetesVersion,
+                agentPoolProfiles: new List<ContainerServiceAgentPoolProfile> {defaultAgentPoolProfile},
+                linuxProfile: linuxProfile,
+                servicePrincipalProfile: spProfile);
+            return managedCluster;
+        }
+
+        /// <summary>
+        /// Fetch SSH public key string
+        /// </summary>
+        /// <param name="sshKeyOrFile">a string representing either the file location, the ssh key pub data or null.</param>
+        /// <returns>SSH public key data</returns>
+        /// <exception cref="ArgumentException">The SSH key or file argument was null and there was no default pub key in path.</exception>
+        protected string GetSshKey(string sshKeyOrFile)
+        {
+            // SSH key was specified as either a file or as key data
+            if (!string.IsNullOrEmpty(SshKeyValue))
```

Comment:
need braces to define the boundaries of the If block

Reply:
Done.